yt_dlp/extractor/tiktok.py

```diff
@@ -314,8 +320,10 @@ def extract_addr(addr, add_meta={}):
             'uploader_id': str_or_none(author_info.get('uid')),
             'uploader_url': user_url,
             'track': music_track,
+            'track_id': str_or_none(music_info.get('id_str')),
             'album': str_or_none(music_info.get('album')) or None,
             'artist': music_author or None,
+            'attributions': contained_music_info,
```

Comment:
I think we should treat contained music and the sound as distinct entities.

1. There is always a sound; contained music is not always present
2. The sound is a TikTok entity with a unique ID. The contained music info just refers to an external track that is present within the sound.

Reply:
Sorry, I still don't understand. Pls note that I don't use tiktok, so internal terms like "sound" mean nothing to me.

Ideally, what we want is that:

* If video is of the music itself, set track_id, artist etc.
* If the video is something else, but music is simply contained in it, set `attributions`.

Is this what is currently happening? The variable names make me think otherwise.

Also, pls add a test for each case.

Reply:
Sorry, I still don't understand. Pls note that I don't use tiktok, so internal terms like "sound" mean nothing to me.

Ideally, what we want is that:

* If video is of the music itself, set track_id, artist etc.
* If the video is something else, but music is simply contained in it, set `attributions`.

This does not seem to be what is currently happening.


For example, you are treating the test <https://www.tiktok.com/@patroxofficial/video/6742501081818877190?langCountry=en> as music but it i just a video that contains the music <https://www.tiktok.com/music/Big-Fun-209649576000286720>